TypeCobol/Compiler/Domain/ProgramSymbolTableBuilder.cs

```diff
@@ -0,0 +1,1034 @@
+                CurrentDataDivisionSection = new DataDivisionSection(Symbol.Flags.FILE, CurrentProgram.FileData);
+            }
+        }
+
+        public override void EndFileSection()
+        {
+            CurrentDataDivisionSection = null;
+        }
+
+        public override void StartGlobalStorageSection(GlobalStorageSectionHeader header)
+        {
+            //No GLOBAL STORAGE SECTION in a procedure, look into CurrentProgram
+            if (CurrentFunction == null)
+            {
+                System.Diagnostics.Debug.Assert(CurrentProgram != null);
+                CurrentDataDivisionSection = new DataDivisionSection(Symbol.Flags.GLOBAL_STORAGE, CurrentProgram.GlobalStorageData);
+            }
+        }
+
+        public override void EndGlobalStorageSection()
+        {
+            CurrentDataDivisionSection = null;
+        }
+
+        public override void StartWorkingStorageSection(WorkingStorageSectionHeader header)
+        {
+            System.Diagnostics.Debug.Assert(CurrentScope != null);
+            CurrentDataDivisionSection = new DataDivisionSection(Symbol.Flags.WORKING_STORAGE, CurrentScope.WorkingStorageData);
+        }
+
+        public override void EndWorkingStorageSection()
+        {
+            CurrentDataDivisionSection = null;
+        }
+
+        public override void StartLocalStorageSection(LocalStorageSectionHeader header)
+        {
+            System.Diagnostics.Debug.Assert(CurrentScope != null);
+            CurrentDataDivisionSection = new DataDivisionSection(Symbol.Flags.LOCAL_STORAGE, CurrentScope.LocalStorageData);
+        }
+
+        public override void EndLocalStorageSection()
+        {
+            CurrentDataDivisionSection = null;
+        }
+
+        public override void StartLinkageSection(LinkageSectionHeader header)
+        {
+            System.Diagnostics.Debug.Assert(CurrentScope != null);
+            CurrentDataDivisionSection = new DataDivisionSection(Symbol.Flags.LINKAGE, CurrentScope.LinkageData);
+        }
+
+        public override void EndLinkageSection()
+        {
+            CurrentDataDivisionSection = null;
+        }
+
+        /// <summary>
+        /// Resolve in the current program linkage section, the given parameter.
+        /// </summary>
+        /// <param name="p">The parameter to resolve.</param>
+        /// <returns>The resolved variable if any, null otherwise</returns>
+        private VariableSymbol ResolveUsingParameter(CallTargetParameter p)
+        {
+            System.Diagnostics.Debug.Assert(p.StorageArea != null);
+            System.Diagnostics.Debug.Assert(p.StorageArea.SymbolReference != null);
+            string parameterName = p.StorageArea.SymbolReference.Name;
+            var parameterCandidateSymbols = this.CurrentProgram.LinkageData.Lookup(parameterName);
+            if (parameterCandidateSymbols != null)
+            {
+                System.Diagnostics.Debug.Assert(parameterCandidateSymbols.Count == 1);
```

Comment:
2 variables in linkage section can have the same name. This must reported as a Diagnostic rather than a Assert.

Reply:
I agree that the `Assert` is not correct here so I temporarily included the check of unicity of lookup results in the enclosing if but this also means that if a parameter is defined 2 or more times in the linkage section we won't have it resolved here. Some checker would have to go over the parameter list (and returning variable) again and verify that everything has been correctly resolved.

Reply:
Parameters resolution have been removed from ProgramSymbolTableBuilder (see 9b44185d641b27af344cc864408c87c88df77f4e). Issue #1730 is dedicated to add all those resolution steps outside of the ProgramSymbolTableBuilder)